activation/post.go

```diff
@@ -444,12 +446,15 @@ func (mgr *PostSetupManager) GenerateProof(ctx context.Context, challenge []byte
 	}
 	mgr.mu.Unlock()
 
-	proof, proofMetadata, err := proving.Generate(ctx, challenge, mgr.cfg.ToConfig(), mgr.logger.Zap(),
+	opts := []proving.OptionFunc{
 		proving.WithDataSource(mgr.cfg.ToConfig(), mgr.id.Bytes(), mgr.commitmentAtxId.Bytes(), mgr.lastOpts.DataDir),
 		proving.WithNonces(mgr.provingOpts.Nonces),
 		proving.WithThreads(mgr.provingOpts.Threads),
 		proving.WithPowFlags(mgr.provingOpts.Flags),
-	)
+	}
+	opts = append(opts, options...)
```

Comment:
This has the drawback that the caller can now overwrite all of these options, which I think is not intended.

We should add at least a `// FIXME:` comment here to address this in the future.

Reply:
I know, but I wouldn't consider it a drawback. Eventually, I see knobs like `threads` and `nonces` and `pow flags` as _options_ .

Reply:
Yes, but we don't want to pass these options n every function call, as they are not expected to change while the node is running 🙂. Thats also why they are set once via the `provingOpts` when initializing this service.

Reply:
Actually changing them while the node is running could be a good UX improvement :thinking: The user could pick how much CPU resources should be spent, maybe use more when the PC is idle etc.